Build fails with Qt 6.6.0

FreeBSD is in the process of updating Qt to 6.6.x and we are seeing a new build failure that wasn't present when building against Qt 6.5.3.

FreeBSD bug report: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=275068
Full build log: https://pkg-status.freebsd.org/package22/data/132amd64-default-foo/2023-11-18_08h04m12s/logs/errors/qucs-s-2.1.0.log

```
FAILED: qucs/diagrams/CMakeFiles/diagrams.dir/graph.cpp.o 
/usr/bin/c++ -DHAVE_CONFIG_H -DQT_NO_DEBUG_OUTPUT -I/wrkdirs/usr/ports/cad/qucs-s/work/.build/qucs/diagrams/diagrams_autogen/include -I/usr/local/include/qt6/QtCore -I/usr/local/include/qt6 -I/usr/local/include/qt6/QtWidgets -I/usr/local/include/qt6/QtGui -I/usr/local/include/qt6/QtDBus -I/usr/local/include/qt6/QtSvg -I/usr/local/include/qt6/QtXml -I/usr/local/include/qt6/QtPrintSupport -I/wrkdirs/usr/ports/cad/qucs-s/work/.build/qucs -I/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs -I/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/components -I/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/diagrams -I/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/dialogs -I/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/paintings -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -isystem /usr/local/include  -DNDEBUG -std=gnu++17 -fPIC -Wall -Wextra -MD -MT qucs/diagrams/CMakeFiles/diagrams.dir/graph.cpp.o -MF qucs/diagrams/CMakeFiles/diagrams.dir/graph.cpp.o.d -o qucs/diagrams/CMakeFiles/diagrams.dir/graph.cpp.o -c /wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/diagrams/graph.cpp
In file included from /wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/diagrams/graph.cpp:17:
In file included from /wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/diagrams/graph.h:22:
/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/diagrams/marker.h:70:23: error: implicit instantiation of undefined template 'std::vector<double>'
  std::vector<double> VarPos;   // values the marker is pointing to
                      ^
/usr/include/c++/v1/iosfwd:260:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
/wrkdirs/usr/ports/cad/qucs-s/work/qucs_s-2.1.0/qucs/diagrams/graph.cpp:126:9: warning: 'setNamedColor' is deprecated: Use fromString() instead. [-Wdeprecated-declarations]
  Color.setNamedColor(n);
        ^
/usr/local/include/qt6/QtGui/qcolor.h:65:5: note: 'setNamedColor' has been explicitly marked deprecated here
    QT_DEPRECATED_VERSION_X_6_6("Use fromString() instead.")
    ^
/usr/local/include/qt6/QtCore/qtdeprecationmarkers.h:175:44: note: expanded from macro 'QT_DEPRECATED_VERSION_X_6_6'
# define QT_DEPRECATED_VERSION_X_6_6(text) QT_DEPRECATED_X(text)
                                           ^
/usr/local/include/qt6/QtCore/qtdeprecationmarkers.h:27:33: note: expanded from macro 'QT_DEPRECATED_X'
#  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
                                ^
/usr/local/include/qt6/QtCore/qcompilerdetection.h:982:36: note: expanded from macro 'Q_DECL_DEPRECATED_X'
#  define Q_DECL_DEPRECATED_X(x) [[deprecated(x)]]
                                   ^
1 warning and 1 error generated.
```

**TODO** list

- [x] Fix missing `vector` include #376 
- [ ] Fix `setNamedColor()` deprecation warning

--- a/qucs/diagrams/marker.h
+++ b/qucs/diagrams/marker.h
@@ -18,6 +18,8 @@
 #ifndef MARKER_H
 #define MARKER_H
 
+#include <vector>
+
 #include "element.h"
 #include "viewpainter.h"
 

